### PR TITLE
Fix focus states for primary navigation, sub navigation and filter links

### PR DESCRIPTION
**Pull request by adamsilver**
## Context

The focus styles are either not there or not very good.

## Changes proposed in this pull request

Add focus states copying the fixes from https://github.com/DFE-Digital/register-trainee-teachers/pull/546

## Link to Trello card

https://trello.com/c/SS0ZVmKm/3413-bug-on-primary-navigation-link-focus

### CODE REVIEW

**Comment by edwardhorsford** [comment]
Will you also add the extra styles for the selected filters?

File: app/frontend/styles/application.scss [hunk 1]
```diff
@@ -9,6 +9,8 @@
 $govuk-font-url-function: frontend-font-url;
 $govuk-image-url-function: frontend-image-url;
 
+$govuk-global-styles: true;
```
Comment (despo) [inline app/frontend/styles/application.scss:12]: What does setting this property do?

File: app/frontend/styles/_primary-navigation.scss [hunk 2]
```diff
@@ -88,6 +99,7 @@
 
       &:before {
         background-color: govuk-colour("black");
+        height: 7px;
```
Comment (despo) [inline app/frontend/styles/_primary-navigation.scss:102]: Shouldn't we be using `rem` instead of pixels so font size is scaled correctly across browsers?

**Review by despo — APPROVED** [review]
Verified fix on review app using Chrome. I have some questions so I can understand why we are doing things a certain way and maybe consider addressing in future PRs, but happy to approve as we can merge this fix.

File: app/frontend/styles/application.scss (hunk 1 above)
Comment (adamsilver) [reply app/frontend/styles/application.scss:12]: It turns on global styles for P tags and A tags. This means you don't have to, for example add `govuk-body` and ‘govuk-link’ respectively onto every link to get the right styles (including focus styles).

It also makes it easy to copy styles from the prototype into production (which we sometimes do) which works the same way.

File: app/frontend/styles/_primary-navigation.scss (hunk 2 above)
Comment (adamsilver) [reply app/frontend/styles/_primary-navigation.scss:102]: I think it's fine - we use pixels in many places that do not impact the font size such as margin, padding and borders.

File: app/frontend/styles/application.scss (hunk 1 above)
Comment (edwardhorsford) [reply app/frontend/styles/application.scss:12]: I might be wary of turning that on. It does catch some things, but in general I've been trying to always use the right classes.

Comment (paulrobertlloyd) [reply app/frontend/styles/application.scss:12]: I suggested turning this on once too… there aspects of the design system that really bug me, and while I totally understand the reasoning, having to add a class to every element to get standard styling is up there on my list of pet peeves!

Comment (paulrobertlloyd) [reply app/frontend/styles/application.scss:12]: In fact, turning on these styles breaks the design of the tile component… link styles and focus state are broken.

Screenshot 2021-02-22 at 10 34 52

Comment (frankieroberto) [reply app/frontend/styles/application.scss:12]: I'm wary of turning it on. The global styles mode has had some bugs in the past (as I don't think it’s routinely tested by the Design System team), and was apparently mainly done to make prototyping easier.

Maybe we can find other ways (like the rubocop rule) to catch elements which don't use the helpers that add the classes?

(I also dislike having to add a class to every element though. Especially as there are some which don't require it, like `<li>`...)

File: app/frontend/styles/_primary-navigation.scss (hunk 2 above)
Comment (edwardhorsford) [reply app/frontend/styles/_primary-navigation.scss:102]: The design system uses `px` for most units - I know for a good reason, but I don't remember them any more. I think all modern browsers will scale them fine.

File: app/frontend/styles/application.scss (hunk 1 above)
Comment (adamsilver) [reply app/frontend/styles/application.scss]: I have removed.

**Review by paulrobertlloyd — APPROVED** [review]

**Review by despo — APPROVED** [review]
Thanks for addressing all feedback and also sorting out your commit log!

**Pull request merged**